packages/tools/devtools/devtools-core/src/data-visualization/SharedTreeVisualizer.ts

```diff
@@ -148,42 +156,53 @@ export function toVisualTree(tree: VisualSharedTreeNode): VisualChildNode {
 }
 
 /**
- * Concatenrate allowed types for `ObjectNodeStoredSchema` and `MapNodeStoredSchema`.
+ * Concatenate allowed types for `ObjectNodeStoredSchema` and `MapNodeStoredSchema`.
  */
-function concatenateTypes(fieldTypes: ReadonlySet<string>): string {
+export function concatenateTypes(fieldTypes: ReadonlySet<string>): string {
 	return [...fieldTypes].join(" | ");
 }
 
 /**
- * Returns the allowed fields & types for the object fields (e.g., `foo : string | number, bar: boolean`)
+ * Properties that describe schema constraints for a field in the tree
  */
-function getObjectAllowedTypes(schema: SimpleObjectNodeSchema): string {
-	const result: string[] = [];
-
-	for (const [fieldKey, treeFieldSimpleSchema] of Object.entries(schema.fields)) {
-		const fieldTypes = treeFieldSimpleSchema.allowedTypes;
-		result.push(`${fieldKey} : ${concatenateTypes(fieldTypes)}`);
-	}
+interface FieldSchemaProperties {
+	allowedTypes: ReadonlySet<string> | undefined;
 
-	return `{ ${result.join(", ")} }`;
+	/**
+	 * Whether the field is required (true) or optional (false).
+	 *
+	 * `undefined` indicates that the field is implicitly required.
+	 * In this case, no requirement information will be displayed by the devtools.
+	 */
+	isRequired: boolean | undefined;
 }
 
 /**
- * Returns the schema & fields of the node.
+ * Processes and visualizes the fields of a verbose tree node.
+ *
+ * @param treeFields - The fields of the tree node to visualize. Can be either an array of VerboseTree (for array nodes) or a Record of field names to VerboseTree (for object/map nodes).
+ * @param treeDefinitions - Map containing all schema definitions for the entire tree structure. Each definition describes the shape and constraints of a particular node type.
+ * @param fieldSchemaProperties - Record mapping field names to their schema properties, including allowed types and whether they are required.
+ * @param visualizeChildData - Callback function to visualize child node data.
+ *
+ * @returns A record mapping field names/indices to their visual tree representations.
  */
 async function visualizeVerboseNodeFields(
-	tree: VerboseTreeNode,
-	treeSchema: SimpleTreeSchema,
+	treeFields: readonly VerboseTree[] | Record<string, VerboseTree>,
+	treeDefinitions: ReadonlyMap<string, SimpleNodeSchema>,
+	fieldSchemaProperties: Record<string, FieldSchemaProperties>,
```

Comment:
Rather than defining your own `FieldSchemaProperties` I think it would make more sense to just use SimpleFieldSchema everywhere. This avoids defining your own redundant type and also makes possible future changes to do things like display more data about the field (like correctly handling identifiers fields or showing metadata) much easier.

Reply:
[31468](https://dev.azure.com/fluidframework/internal/_sprints/taskboard/Team%20Rohan/internal/Team%20Rohan%20-%202025.Cummins?workitem=31468)

Created a follow-up task for this. Will request the review on subsequent refactor task. Thanks!